lib/views/survey/survey_field.dart

```diff
@@ -0,0 +1,375 @@
+// ignore_for_file: avoid_renaming_method_parameters, type_annotate_public_apis
+
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:thc/models/theme.dart';
+import 'package:thc/views/survey/survey_questions.dart';
+import 'package:thc/views/survey/survey_screen.dart';
+
+/// {@template views.survey.SurveyField}
+/// A widget that displays a survey question.
+/// {@endtemplate}
+class SurveyField extends StatelessWidget {
+  /// {@macro views.survey.SurveyField}
+  const SurveyField(this.record, this.update, {super.key});
+
+  /// {@macro views.survey.SurveyRecord}
+  final SurveyRecord record;
+
+  /// {@template ValueChanged}
+  /// For a very long time, I had no idea what [ValueChanged] was;
+  /// then I realized it's just a function signature.
+  ///
+  /// 2 ways to write the same thing:
+  /// ```dart
+  /// final ValueChanged<int> onChanged;
+  /// final void Function(int) onChanged;
+  /// ```
+  /// {@endtemplate}
+  final ValueChanged<dynamic> update;
+
+  @override
+  Widget build(BuildContext context) {
+    final builder = SurveyBuilder.fromRecord(record);
+    final question = _QuestionText(record.question);
+
+    Widget answer = builder.buildAnswer(context, update, record.question, record.cleanAnswer);
+    if (record.question is MultipleChoice) {
+      answer = _MultipleChoiceTheme(child: answer);
+    }
+
+    return _ErrorBox(
+      valid: record.valid || !context.watch<SurveyValidation>().state,
+      child: builder.layout(context, question, answer),
+    );
+  }
```

Comment:
This was originally a [192-line build method](https://github.com/theheartcenter-one/thc/blob/e2be88abd8544ebae750d1874b95a6d16a6ee6d5/lib/views/survey/survey_screen.dart#L412), but then I did a whole bunch of refactoring today 🙂

Reply:
Thanks for the efforts!